cpp/modmesh/toggle/SerializableProfiler.hpp

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+/*
+ * Copyright (c) 2024, Chun-Shih Chang <austin20463@gmail.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ * - Neither the name of the copyright holder nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <modmesh/toggle/RadixTree.hpp>
+#include <modmesh/serialization/SerializableItem.hpp>
+
+namespace modmesh
+{
+
+class SerializableRadixTreeNode : SerializableItem
+{
+
+public:
+
+    using child_list_type = std::vector<SerializableRadixTreeNode>;
+    using key_type = typename RadixTree<CallerProfile>::key_type;
+
+    SerializableRadixTreeNode() = default;
+    SerializableRadixTreeNode(SerializableRadixTreeNode const &) = default;
+    SerializableRadixTreeNode(SerializableRadixTreeNode &&) = default;
+    SerializableRadixTreeNode & operator=(SerializableRadixTreeNode const &) = default;
+    SerializableRadixTreeNode & operator=(SerializableRadixTreeNode &&) = default;
+    ~SerializableRadixTreeNode() = default;
+    SerializableRadixTreeNode(const RadixTreeNode<CallerProfile> * node)
+        : m_key(node->key())
+        , m_name(node->name())
+        , m_total_time(node->data().stable_total_time)
+        , m_call_count(node->data().stable_call_count)
+    {
+        for (const auto & child : node->children())
+        {
+            if (child->data().stable_call_count > 0)
```

Comment:
The node is valid only if the root function has completed at least once.

Reply:
How about putting the state in the serialization result? such as `"status: "processing"`, `"status": "done"`.

Reply:
> How about putting the state in the serialization result? such as `"status: "processing"`, `"status": "done"`.

While this is certainly doable, I'm unsure if it's necessary. When deserializing the result, the `status` may not be particularly useful since the profiler should already reflect the state before or after processing. However, including the `status` in the serialization result could make it easier to monitor the serializer.

Reply:
We do not need to make a decision now.  It can be done if a use case surfaces.  That is, we do not need to do anything now.